packages/sdk/typescript/human-protocol-sdk/test/escrow.test.ts

```
@@ -0,0 +1,1368 @@
+      escrowClient.escrowContract.setup.mockReturnValue(true);
+
+      await escrowClient.createAndSetupEscrow(
+        tokenAddress,
+        trustedHandlers,
+        escrowConfig
+      );
+
+      expect(escrowClient.createEscrow).toHaveBeenCalledWith(
+        tokenAddress,
+        trustedHandlers
+      );
+      expect(escrowClient.escrowContract.setup).toHaveBeenCalledWith(
+        ethers.constants.AddressZero,
+        ethers.constants.AddressZero,
+        BigNumber.from(50),
+        BigNumber.from(50),
+        VALID_URL,
+        FAKE_HASH
+      );
+    });
+
+    test('should throw an error if the setup escrow fails', async () => {
+      const escrowConfig = {
+        recordingOracle: ethers.constants.AddressZero,
+        reputationOracle: ethers.constants.AddressZero,
+        recordingOracleFee: BigNumber.from(50),
+        reputationOracleFee: BigNumber.from(50),
+        manifestUrl: VALID_URL,
+        hash: FAKE_HASH,
+      };
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.escrowContract.setup.mockRejectedValueOnce(new Error());
+
+      await expect(
+        escrowClient.setup(ethers.constants.AddressZero, escrowConfig)
+      ).rejects.toThrow();
+
+      expect(escrowClient.escrowContract.setup).toHaveBeenCalledWith(
+        ethers.constants.AddressZero,
+        ethers.constants.AddressZero,
+        BigNumber.from(50),
+        BigNumber.from(50),
+        VALID_URL,
+        FAKE_HASH
+      );
+    });
+  });
+
+  describe('fund', () => {
+    test('should throw an error if escrowAddress is an invalid address', async () => {
+      const invalidAddress = FAKE_ADDRESS;
+      const amount = BigNumber.from(10);
+
+      await expect(escrowClient.fund(invalidAddress, amount)).rejects.toThrow(
+        ErrorInvalidEscrowAddressProvided
+      );
+    });
+
+    test('should throw an error if an hasEscrow return false', async () => {
```

fix writing:
'should throw an error if hasEscrow returns false'